libs/platform/src/lib/components/form/combobox/commons/base-combobox.ts

```diff
@@ -0,0 +1,749 @@
+import {
+    AfterViewInit,
+    ChangeDetectorRef,
+    ContentChildren,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Inject,
+    Injector,
+    Input,
+    OnDestroy,
+    Optional,
+    Output,
+    QueryList,
+    Self,
+    TemplateRef,
+    ViewChildren
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { DOWN_ARROW, ESCAPE, UP_ARROW } from '@angular/cdk/keycodes';
+
+import { fromEvent, isObservable, Observable, Subject, Subscription } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import {
+    DIALOG_CONFIG,
+    DialogConfig,
+    DynamicComponentService,
+    KeyUtil,
+    ListItemDirective,
+    MenuKeyboardService,
+    MobileModeConfig, TemplateDirective
+} from '@fundamental-ngx/core';
+import {
+    ArrayComboBoxDataSource,
+    ComboBoxDataSource,
+    isDataSource,
+    isOptionItem,
+    MatchingBy,
+    ObservableComboBoxDataSource,
+    OptionItem
+} from '../../../../domain';
+
+import { isFunction, isJsObject, isString } from '../../../../utils/lang';
+import { CollectionBaseInput } from '../../collection-base.input';
+import { ComboboxComponent } from '../combobox/combobox.component';
+import { ComboboxMobileComponent } from '../combobox-mobile/combobox/combobox-mobile.component';
+import { COMBOBOX_COMPONENT } from '../combobox.interface';
+import { ComboboxConfig, MatchingStrategy } from '../combobox.config';
+import { ContentDensity } from '../../form-control';
+
+export type TextAlignment = 'left' | 'right';
+export type FdpComboBoxDataSource<T> = ComboBoxDataSource<T> | Observable<T[]> | T[];
+
+export class ComboboxSelectionChangeEvent {
+    constructor(
+        public source: ComboboxComponent,
+        public payload: any // Contains selected item
+    ) {}
+}
+
+@Directive()
+export abstract class BaseCombobox extends CollectionBaseInput implements AfterViewInit, OnDestroy {
+    /** Provides maximum height for the optionPanel */
+    @Input()
+    maxHeight = '250px';
+
+    @Input()
+    get dataSource(): FdpComboBoxDataSource<any> {
+        return this._dataSource;
+    }
+
+    set dataSource(value: FdpComboBoxDataSource<any>) {
+        if (value) {
+            this._initializeDataSource(value);
+        }
+    }
+
+    /** Whether the autocomplete should be enabled; Enabled by default */
+    @Input()
+    autoComplete = true;
+
+    /**
+     * content Density of element. 'cozy' | 'compact'
+     */
+    @Input()
+    set contentDensity(contentDensity: ContentDensity) {
+        this._contentDensity = contentDensity;
+        this.isCompact = contentDensity === 'compact';
```

Comment:
for consistency why isn't this name compact?

Reply:
@stefanoScalzo I believe you need to catch up with Platform 😋
`contentDensity` is platforms approach on naming part of the API related to setting the component in cozy, compact.... It's more universal and flexible than using `compact: boolean` which limits us to two modes.